src/Orleans.Runtime/Catalog/Catalog.cs

```diff
@@ -248,14 +249,17 @@ public void UnregisterSystemTarget(ISystemTarget target)
         /// <returns></returns>
         public IGrainContext GetOrCreateActivation(
             in GrainId grainId,
-            Dictionary<string, object> requestContextData)
+            Dictionary<string, object> requestContextData,
+            MigrationContext rehydrationContext)
         {
             if (TryGetGrainContext(grainId, out var result))
             {
+                rehydrationContext?.Dispose();
```

Comment:
I think we should at least log a warning (maybe throw?) if `rehydrationContext` isn't null in this case